integration-tests/cucumber-tests-platform-publiclighting/src/test/java/org/opensmartgridplatform/cucumber/platform/publiclighting/mocks/oslpdevice/DeviceState.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright 2020 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.cucumber.platform.publiclighting.mocks.oslpdevice;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ConcurrentMap;
+
+import org.opensmartgridplatform.cucumber.core.ScenarioContext;
+import org.opensmartgridplatform.cucumber.platform.PlatformKeys;
+import org.opensmartgridplatform.oslp.Oslp;
+import org.opensmartgridplatform.shared.infra.jms.MessageType;
+
+import lombok.Getter;
+import lombok.Setter;
+
+public class DeviceState {
+
+    @Getter
+    private final String deviceUID;
+    @Getter @Setter
+    private Integer sequenceNumber = 0;
+
+    private static final Integer SEQUENCE_NUMBER_MAXIMUM = 65535;
+
+    private final ConcurrentMap<MessageType, ConcurrentLinkedQueue<Oslp.Message>> mockedResponsesMap = new ConcurrentHashMap<>();
+    private final ConcurrentMap<MessageType, ConcurrentLinkedQueue<Oslp.Message>> receivedRequestsMap = new ConcurrentHashMap<>();
+
+    public DeviceState(final String deviceUID) {
+        this.deviceUID = deviceUID;
+    }
+
+    public Oslp.Message getMockedResponse(final MessageType messageType) throws DeviceSimulatorException {
+        return this.getFromMap(this.mockedResponsesMap, messageType);
+    }
+
+    public Oslp.Message getReceivedRequest(final MessageType messageType) throws DeviceSimulatorException {
+        return this.getFromMap(this.receivedRequestsMap, messageType);
+    }
+
+    public void addMockedResponse(final MessageType messageType, final Oslp.Message message) {
+        this.getQueue(this.mockedResponsesMap, messageType).add(message);
+    }
+
+    public void addReceivedRequest(final MessageType messageType, final Oslp.Message message) {
+        this.getQueue(this.receivedRequestsMap, messageType).add(message);
```

Comment:
I go with you for getNextMessage, but not with addNextMessage. Because when you add a message, getNextMessage can return a different message. I changed the getters to poll<xx>

Reply:
Ha, I had considered something with poll as a suggestion but rejected it because I felt it was too much about a non-essential implementation aspect. I am OK if you stick with the poll-based naming though, as I think it isn't a very clear cut case which is better.